### ISSUE TITLE
schema typedef error: definition run_on_first does not exist

### ISSUE THREAD

**Issue by David-Lor** [alexashley/kong-plugin-path-prefix#5]
Using this plugin in Kong 2.0.3, got the following error when Kong starts up:

```
kong_1  | 2021/07/08 13:20:03 [error] 1#0: init_by_lua error: /usr/local/share/lua/5.1/kong/tools/utils.lua:620: error loading module 'kong.plugins.path-prefix.schema':
kong_1  | ...arocks/share/lua/5.1/kong/plugins/path-prefix/schema.lua:10: schema typedef error: definition run_on_first does not exist
kong_1  | stack traceback:
kong_1  | 	[C]: in function 'error'
kong_1  | 	/usr/local/share/lua/5.1/kong/db/schema/typedefs.lua:515: in function '__index'
kong_1  | 	...arocks/share/lua/5.1/kong/plugins/path-prefix/schema.lua:10: in main chunk
kong_1  | 	[C]: at 0x7fac4572c0d0
kong_1  | 	[C]: in function 'xpcall'
kong_1  | 	/usr/local/share/lua/5.1/kong/tools/utils.lua:611: in function 'load_module_if_exists'
kong_1  | 	/usr/local/share/lua/5.1/kong/db/schema/plugin_loader.lua:183: in function 'load_subschema'
kong_1  | 	/usr/local/share/lua/5.1/kong/db/dao/plugins.lua:232: in function 'load_plugin'
kong_1  | 	/usr/local/share/lua/5.1/kong/db/dao/plugins.lua:275: in function 'load_plugin_schemas'
kong_1  | 	/usr/local/share/lua/5.1/kong/init.lua:425: in function 'init'
kong_1  | 	init_by_lua:3: in main chunk
kong_1  | stack traceback:
kong_1  | 	[C]: in function 'error'
kong_1  | 	/usr/local/share/lua/5.1/kong/tools/utils.lua:620: in function 'load_module_if_exists'
kong_1  | 	/usr/local/share/lua/5.1/kong/db/schema/plugin_loader.lua:183: in function 'load_subschema'
kong_1  | 	/usr/local/share/lua/5.1/kong/db/dao/plugins.lua:232: in function 'load_plugin'
kong_1  | 	/usr/local/share/lua/5.1/kong/db/dao/plugins.lua:275: in function 'load_plugin_schemas'
kong_1  | 	/usr/local/share/lua/5.1/kong/init.lua:425: in function 'init'
kong_1  | 	init_by_lua:3: in main chunk
kong_1  | nginx: [error] init_by_lua error: /usr/local/share/lua/5.1/kong/tools/utils.lua:620: error loading module 'kong.plugins.path-prefix.schema':
kong_1  | ...arocks/share/lua/5.1/kong/plugins/path-prefix/schema.lua:10: schema typedef error: definition run_on_first does not exist
kong_1  | stack traceback:
kong_1  | 	[C]: in function 'error'
kong_1  | 	/usr/local/share/lua/5.1/kong/db/schema/typedefs.lua:515: in function '__index'
kong_1  | 	...arocks/share/lua/5.1/kong/plugins/path-prefix/schema.lua:10: in main chunk
kong_1  | 	[C]: at 0x7fac4572c0d0
kong_1  | 	[C]: in function 'xpcall'
kong_1  | 	/usr/local/share/lua/5.1/kong/tools/utils.lua:611: in function 'load_module_if_exists'
kong_1  | 	/usr/local/share/lua/5.1/kong/db/schema/plugin_loader.lua:183: in function 'load_subschema'
kong_1  | 	/usr/local/share/lua/5.1/kong/db/dao/plugins.lua:232: in function 'load_plugin'
kong_1  | 	/usr/local/share/lua/5.1/kong/db/dao/plugins.lua:275: in function 'load_plugin_schemas'
kong_1  | 	/usr/local/share/lua/5.1/kong/init.lua:425: in function 'init'
kong_1  | 	init_by_lua:3: in main chunk
kong_1  | stack traceback:
kong_1  | 	[C]: in function 'error'
kong_1  | 	/usr/local/share/lua/5.1/kong/tools/utils.lua:620: in function 'load_module_if_exists'
kong_1  | 	/usr/local/share/lua/5.1/kong/db/schema/plugin_loader.lua:183: in function 'load_subschema'
kong_1  | 	/usr/local/share/lua/5.1/kong/db/dao/plugins.lua:232: in function 'load_plugin'
kong_1  | 	/usr/local/share/lua/5.1/kong/db/dao/plugins.lua:275: in function 'load_plugin_schemas'
kong_1  | 	/usr/local/share/lua/5.1/kong/init.lua:425: in function 'init'
kong_1  | 	init_by_lua:3: in main chunk
```

If I modify the `schema.lua` file to its original form (as somebody already did in this fork: https://github.com/famkampm/kong-plugin-path-prefix/commit/f5a5ba473f03c63b134093391982152023119101), the plugin works great (and I must say thank you for it! :smile_cat: ).

I guess the problem is with the `schema.lua` file structure, which maybe changed for Kong 1.x and then again for Kong 2.x? (as I can see in commit https://github.com/alexashley/kong-plugin-path-prefix/commit/3ba5a6c4f27b1c1e7011528cc05909b2e5b20fcb).

The working schema.lua would be this:
```lua
return {
  name = "path-prefix",
  fields = {
    {
      config = {
        type = "record",
        fields = {
          { path_prefix = {type = "string", required = true} },
          { escape = {type = "boolean", default = true} },
        }
      }
    }
  }
}
```

If that is the problem, maybe could consider:

- Warning about it on README?
- Supporting both formats? (not really sure how)

## Scenario

Scenario used for testing this and getting the shown error:

docker_compose.yml
```yaml
version: "3"

services:
    kong:
        build:
            context: ./
        environment:
            KONG_PLUGINS: "bundled,path-prefix"
            KONG_DATABASE: "off"
            KONG_DECLARATIVE_CONFIG: "/kong.yml"
        volumes:
            - "./kong.yml:/kong.yml:ro"
```

Dockerfile
```dockerfile
FROM kong:2.0.3

USER root
RUN apk update && apk add git

USER kong
RUN (cd /tmp && git clone https://github.com/alexashley/kong-plugin-path-prefix) && \
    (cd /tmp/kong-plugin-path-prefix && luarocks make --local "kong-plugin-path-prefix-0.1.0-1.rockspec")
```

kong.yml
```yaml
_format_version: "1.1"

services:
  - name: service
    plugins:
      - path-prefix:
          config:
            path_prefix: "/foo"
```

**Comment by alexashley** [comment]
Hi @David-Lor, thanks for the detailed report and my apologies for the delay in responding. As you might have guessed from the activity on this repository, I don't use Kong in my day to day any longer and haven't kept this plugin up to date. 

I opened #6 with a PR to remove the `run_on` section of the config along with some housekeeping. 
It looks like `run_on` was removed in 2.x, although the typedefs still seem useful so I left them as is. The very basic tests I have were passing for Kong 2.5.0, but let me know if you're still seeing an issue.
